consult.el

```diff
@@ -932,6 +936,70 @@ Depending on the selected item OPEN-BUFFER, OPEN-FILE or OPEN-BOOKMARK will be u
   (interactive)
   (consult--buffer #'switch-to-buffer #'find-file #'bookmark-jump))
 
+;;;###autoload
+(defun consult-kmacro (arg)
+  "Run a chosen keyboard macro.  With prefix ARG, run the macro that many times.
+
+Macros containing mouse clicks aren't displayed."
+  (interactive "p")
+  (if (not (or last-kbd-macro kmacro-ring))
+      (user-error "No keyboard macros defined")
+    (let* ((numbered-kmacros
+            (seq-uniq (let (result (index -1))
+                        (dolist ( kmacro (cons (if (listp last-kbd-macro)
+                                                   last-kbd-macro
+                                                 (list last-kbd-macro
+                                                       kmacro-counter
+                                                       kmacro-counter-format))
+                                               kmacro-ring)
+                                  (nreverse result))
+                          (cl-incf index)
+                          (unless (seq-some #'mouse-event-p (car kmacro))
```

Comment:
I would prefer seq-filter, followed by mapcar, followed by seq-uniq. Usually I juggle a bit around until it feels right. This is probably different from writing cl-loops. If you start functional and decompose everything, it stays pretty readable and is often not worse than cl-loop. I don't know how you feel about this?